include/dali/core/tensor_shape.h

```diff
@@ -894,8 +892,7 @@ struct TensorListShape : TensorListShapeBase<TensorListShape<sample_ndim>, sampl
   TensorListShape &operator=(const TensorListShape &) = default;
   TensorListShape &operator=(TensorListShape &&other) {
     shapes = std::move(other.shapes);
-    nsamples = other.size();
-    other.nsamples = 0;
+    nsamples = std::exchange(other.nsamples, 0);
```

Comment:
IIUC, previously the nsamples wasn't moved, but passed by value, so this changes the behaviour. Is that intentional?

Reply:
As above